### PR TITLE
Master-port #49860 selinux support for file.manage

### PR DESCRIPTION
**Pull request by mchugh19**
Master port of #49860
This looks good to me, but happy to get more eyes on it. See conversation in #49860 for more information

### What does this PR do?
Recursion not supported for persist=True on file.get_selinux_context execution module

Add selinux functionality to file.managed
```
/tmp/selinux.test
  file.managed:
    - user: root
    - selinux:
        seuser:  system_u
        serole: object_r
        setype: system_conf_t
        seranage: s0
```
```
# salt-call --local state.sls selinux test=True
local:
----------
          ID: check_selinux
    Function: file.managed
        Name: /tmp/selinux.test
      Result: None
     Comment: File /tmp/selinux.test not updated
     Started: 10:54:39.418112
    Duration: 17.911 ms
     Changes:
              ----------
              selinux context:
                  ----------
                  New:
                      User: system_u Type: system_conf_t
                  Old:
                      User: unconfined_u Type: user_tmp_t

Summary for local
------------
Succeeded: 1 (unchanged=1, changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  17.911 ms
```

### What issues does this PR fix or reference?
#40703 and possibly #1349

### New Behavior
Uses file.get_selinux_context and file.set_selinux_context to determine if selinux updates are needed. As mentioned in a ticket #1349, set_selinux_context uses the chcon command, which might not be persistent. Feedback welcome.

### Tests written?
Yes - but could always use more eyes

### CODE REVIEW

File: salt/modules/file.py [hunk 1]
```diff
@@ -4531,8 +4544,13 @@ def _add_to_matches(found, line, match_type, value, matched):
     return None
 
 
-def check_perms(name, ret, user, group, mode, attrs=None, follow_symlinks=False):
+def check_perms(name, ret, user, group, mode, attrs=None, follow_symlinks=False, seuser=None,
+                serole=None, setype=None, serange=None):
     '''
+    .. versionchanged:: Neon
```
Comment (twangboy) [inline salt/modules/file.py]: Sodium

File: salt/modules/file.py [hunk 2]
```diff
@@ -4809,10 +4924,18 @@ def check_managed_changes(
...
         keep_mode=False,
+        seuser=None,
+        serole=None,
+        setype=None,
+        serange=None,
         **kwargs):
     '''
     Return a dictionary of what changes need to be made for a file
 
+    .. versionchanged:: Neon
```
Comment (twangboy) [inline salt/modules/file.py]: Sodium

File: salt/modules/file.py [hunk 3]
```diff
@@ -4925,6 +5053,26 @@ def check_file_meta(
...
+
+    setype
+        selinux type attribute
+
+        .. versionadded:: Neon
+
+    serange
+        selinux range attribute
+
+        .. versionadded:: Neon
```
Comment (twangboy) [inline salt/modules/file.py]: These should all be Sodium

File: salt/modules/file.py [hunk 4]
```diff
@@ -5247,6 +5420,26 @@ def manage_file(name,
...
+
```
Comment (twangboy) [inline salt/modules/file.py]: Sodium

File: salt/states/file.py [hunk 5]
```diff
@@ -2685,6 +2686,22 @@ def managed(name,
...
+            /tmp/selinux.test
+              file.managed:
+                - user: root
+                - selinux:
+                    seuser: system_u
+                    serole: object_r
+                    setype: system_conf_t
+                    seranage: s0
+
+        .. versionadded:: Neon
```
Comment (twangboy) [inline salt/states/file.py]: Sodium

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/file.py [hunk 6]
```diff
@@ -1294,7 +1293,7 @@ def psed(path,
     after = six.text_type(after)
     before = _sed_esc(before, escape_all)
     # The pattern to replace with does not need to be escaped!!!
-    #after = _sed_esc(after, escape_all)
+    # after = _sed_esc(after, escape_all)
```
[the remark this answers is not in the document]
Comment (mchugh19) [reply salt/modules/file.py]: I'm not responsible for the commented code, or the comment. But I've removed the dead code as suggested.

File: salt/modules/file.py (hunk 1 above)
Comment (mchugh19) [reply salt/modules/file.py]: Done

File: salt/modules/file.py (hunk 2 above)
Comment (mchugh19) [reply salt/modules/file.py]: Done

File: salt/modules/file.py (hunk 3 above)
Comment (mchugh19) [reply salt/modules/file.py]: Done

File: salt/modules/file.py (hunk 4 above)
Comment (mchugh19) [reply salt/modules/file.py]: Done

File: salt/states/file.py (hunk 5 above)
Comment (mchugh19) [reply salt/states/file.py]: Done

**Review by twangboy — DISMISSED** [review]

**Comment by twangboy** [comment]
Hey, @mchugh19 

Looks like this needs to be rebased and have pre-commit run on it. Do you mind doing that?

https://docs.saltstack.com/en/latest/topics/development/contributing.html#quickstart

**Comment by mchugh19** [comment]
@twangboy 
> Looks like this needs to be rebased and have pre-commit run on it. Do you mind doing that?

I'm not sure what you mean. These files have been run through isort and black already. I tried manually running `pre-commit run --files filenames` but there weren't any changes. and there don't appear to be any conflicts with master. What needs rebasing?

**Comment by waynew** [comment]
I'm not sure either 😂 

Pre-commit looks fine and is passing, but there are some other failures that look to be related to this PR. Can you take a look?

**Comment by mchugh19** [comment]
tests were missing a `cmd.run_all` mapping. Not sure how it ever worked before, or why it only failed on centos (maybe selinux is only enabled on redhat?). Should be good to go now though.

**Comment by mchugh19** [comment]
re-run pr-windows2019-py3

**Comment by mchugh19** [comment]
re-run pr-windows2016-py3

**Comment by mchugh19** [comment]
re-run pr-centos7-py3

**Comment by mchugh19** [comment]
Ugh. Centos is stuck on 
```
File "/tmp/kitchen/testing/tests/integration/states/test_docker_network.py", line 133, in tearDownClass
    if process.returncode != 0 and "No such image" not in output:
TypeError: a bytes-like object is required, not 'str'
```

**Review by dwoz — APPROVED** [review]

**Pull request merged**